lib/plugin.js

```diff
@@ -37,62 +37,60 @@ const loadScript = function(url, options) {
 
 const scripts = [];
 const loadingScripts = {};
-const getScript = (plugin.getScript = function(url, callback) {
+const getScript = (plugin.getScript = async function(url, callback) {
   if (callback == null) { callback = function() {}; }
   if (Array.from(scripts).includes(url)) {
     return callback();
   } else {
-    return loadScript(url)
-      .done(function() {
-        scripts.push(url);
-        return callback();}).fail(function(_jqXHR, _textStatus, err) {
-        console.log('getScript: Failed to load:', url, err);
-        return callback();
-    });
+    await new Promise(function(resolve, reject) {
+      loadScript(url)
+        .done(function() {
+          scripts.push(url);
+          return resolve();
+        }).fail(function(_jqXHR, _textStatus, err) {
+          console.log('getScript: Failed to load:', url, err);
+          return resolve();
+      });
+    })
+    return callback();
   }
 });
 
-plugin.renderFrom = function(notifIndex) {
+plugin.renderFrom = async function(notifIndex) {
   const $items = $(".item").slice(notifIndex);
   
   // console.log "notifIndex", notifIndex, "about to render", $items.toArray()
-  let promise = Promise.resolve();
-  var emitNextItem = function(itemElems) {
-    if (itemElems.length === 0) { return promise; }
-    const itemElem = itemElems.shift();
+
+  for (const itemElem of $items.toArray()) {
     const $item = $(itemElem);
     if (!$item.hasClass('textEditing')) {
       const item = $item.data('item');
-      promise = promise.then(() => new Promise(function(resolve, reject) {
+      try {
         $item.off();
-        return plugin.emit($item.empty(), item, () => resolve());
-      }));
+        await plugin.emit($item.empty(), item)
```

Comment:
Yeah, the await is a pretty direct replacement for the callback w/ resolve, the main difference is an error thrown in `plugin.emit` would now surface here and get caught. I left the callbacks on all of these functions available but optional, since they are exported, but with promises flowing through all of them now we only have to do the `new Promise(function (resolve...` thing at the edges where this file interacts with external callback functions.